pkg/api/v1/resources/common/kubernetes/config_map_multi_cluster_client.sk.go

```diff
@@ -0,0 +1,115 @@
+// Code generated by solo-kit. DO NOT EDIT.
+
+package kubernetes
+
+import (
+	"sync"
+
+	"github.com/solo-io/go-utils/errors"
+	"github.com/solo-io/solo-kit/pkg/api/v1/clients"
+	"github.com/solo-io/solo-kit/pkg/api/v1/clients/factory"
+	"github.com/solo-io/solo-kit/pkg/api/v1/clients/wrapper"
+	"github.com/solo-io/solo-kit/pkg/multicluster"
+	"k8s.io/client-go/rest"
+)
+
+type ConfigMapMultiClusterClient interface {
+	multicluster.ClusterHandler
+	ConfigMapInterface
+}
+
+type configMapMultiClusterClient struct {
+	clients       map[string]ConfigMapClient
+	clientAccess  sync.RWMutex
+	aggregator    wrapper.WatchAggregator
+	factoryGetter factory.ResourceClientFactoryGetter
```

Comment:
I don't love that this requires the user to implement a FactoryGetter, as it will often amount to a wrapper for the shared/core cache manager, but this cleans up the generated code quite a bit.